tests/Functional/Query/QueryBuilderTest.php

```diff
@@ -332,6 +334,67 @@ public function testUnionAndAddUnionWorksWithQueryBuilderPartsAndReturnsExpected
         self::assertSame($expectedRows, $qb->executeQuery()->fetchAllAssociative());
     }
 
+    public function testNotSupportedCTE(): void
+    {
+        if ($this->connection->getDatabasePlatform()->supportsCTEs()) {
+            self::markTestSkipped('The database platform does support CTE.');
+        }
+
+        $qb = $this->connection->createQueryBuilder();
+
+        $cteQueryBuilder1 = $this->connection->createQueryBuilder();
+        $cteQueryBuilder1->select('id')
+            ->from('for_update');
+
+        self::expectException(QueryException::class);
+        self::expectExceptionMessage('CTEs are not supported by this database platform.');
+
+        $qb->with('filtered_for_update', $cteQueryBuilder1);
+    }
+
+    public function testWithNamedParameterCTE(): void
+    {
+        if (! $this->connection->getDatabasePlatform()->supportsCTEs()) {
+            self::markTestSkipped('The database platform does not support CTE.');
+        }
```

Comment:
In tests, please don't ever ask the platform if it supports a certain feature. The test needs to **know** this by itself, otherwise we produce a green build because of a bug in the implementation.

In this particular case: Skip the test if the platform is MySQL 5.7.